### ISSUE TITLE
[bug] sui-test-validator does not work

### ISSUE THREAD

**Issue by icodezjb** [MystenLabs/sui#5809]
`sui`: `devnet-v0.14.0`
`OS`: ubuntu 20.04
`cmd`: `sui-test-validator`
`expect`:  `4 validators + 1 full node` testnet should work.
`error`: 
```
2022-11-03T06:32:26.505723Z  INFO sui_cluster_test::faucet: Use local faucet
2022-11-03T06:32:37.662934Z ERROR node{name=0x24fe33c99dc8dc906e480b74dfcb1a3df41af0c6}: narwhal_consensus::bullshark: The store does not contain the parent of M6j8uvECx3hTtL94: C166(hstIluX0uuFxUrSjTSUCTXjOadA4prItEXp23IkLK7Vgq9LbWSU8KeXixoSMXMjcFDRKyhQeSgZID1aJdXsyleSsHP7yu6io7hcWtewb5qQvuD31P6WKNwOPKDuIJqUO, irTovHjrXvNjFAs/, E0): Missing item digest=7qdzheXZkt1JU4O9dHnw9sBmaGFaB1rPVFN2fEPOBLw=
2022-11-03T06:33:38.713634Z ERROR node{name=0x93d32abb91810430982c85a98c6bfcc551fd6225}: narwhal_consensus::bullshark: The store does not contain the parent of 7RBOLrzrLgQ9c8xT: C918(q+kxsnIEXHhlRTd3DkOZvx6PIaUyzxPaNZbbQJ2E3QzeZSy9ooZXWznl3KAFHo1RA2naZQxXBCqJfOKWMR87MoM0hoVg8NPfEwL9gL5jApW1m3ToMkQRmS8u/PfwEAaM, 9TLtTOWqn+60MRiV, E0): Missing item digest=HPrIJION8B3drd9H04qXqqkrplmVk5ZTZC+Xju6gQnc=
2022-11-03T06:33:52.347799Z ERROR node{name=0x93d32abb91810430982c85a98c6bfcc551fd6225}: narwhal_consensus::bullshark: The store does not contain the parent of hV7GGnZ7RdqwtOJX: C1088(jyNlwfC5VH7wGMc/2eFEbKR4A8VGw7m9hfPmFjoAl18GpbOx/n6oMM6gEPZtdeARCAMmmVbIwPBM33NLFHoAZG5RzPJbqeldl3lo6IiyjZTagiinW6dxj7l6PVztEfFm, KFC4Q+sC9sAu7nP5, E0): Missing item digest=F4/ZnsPCbPEmipN43l80TVGJQPobUpgI9hWRt/xZemk=
2022-11-03T06:33:52.348281Z ERROR node{name=0xb31ba1d39aad808e5f67fb889c4c5da12989de0d}: narwhal_consensus::bullshark: The store does not contain the parent of hV7GGnZ7RdqwtOJX: C1088(jyNlwfC5VH7wGMc/2eFEbKR4A8VGw7m9hfPmFjoAl18GpbOx/n6oMM6gEPZtdeARCAMmmVbIwPBM33NLFHoAZG5RzPJbqeldl3lo6IiyjZTagiinW6dxj7l6PVztEfFm, KFC4Q+sC9sAu7nP5, E0): Missing item digest=F4/ZnsPCbPEmipN43l80TVGJQPobUpgI9hWRt/xZemk=
2022-11-03T06:33:52.348358Z ERROR node{name=0x3d14c61174ec1b9cb34b8a1628d4c7fe389b0b0b}: narwhal_consensus::bullshark: The store does not contain the parent of hV7GGnZ7RdqwtOJX: C1088(jyNlwfC5VH7wGMc/2eFEbKR4A8VGw7m9hfPmFjoAl18GpbOx/n6oMM6gEPZtdeARCAMmmVbIwPBM33NLFHoAZG5RzPJbqeldl3lo6IiyjZTagiinW6dxj7l6PVztEfFm, KFC4Q+sC9sAu7nP5, E0): Missing item digest=F4/ZnsPCbPEmipN43l80TVGJQPobUpgI9hWRt/xZemk=
2022-11-03T06:34:11.438096Z ERROR node{name=0x24fe33c99dc8dc906e480b74dfcb1a3df41af0c6}: narwhal_consensus::bullshark: The store does not contain the parent of i0rVZTTfOrhyZXWg: C1310(q+kxsnIEXHhlRTd3DkOZvx6PIaUyzxPaNZbbQJ2E3QzeZSy9ooZXWznl3KAFHo1RA2naZQxXBCqJfOKWMR87MoM0hoVg8NPfEwL9gL5jApW1m3ToMkQRmS8u/PfwEAaM, yjT+gfSMgP3mkMMZ, E0): Missing item digest=U01zZYIUj6YuQIVaH+o/aHIwoKW158dY0ptBESdO9I0=
2022-11-03T06:34:11.448162Z ERROR node{name=0x3d14c61174ec1b9cb34b8a1628d4c7fe389b0b0b}: narwhal_consensus::bullshark: The store does not contain the parent of i0rVZTTfOrhyZXWg: C1310(q+kxsnIEXHhlRTd3DkOZvx6PIaUyzxPaNZbbQJ2E3QzeZSy9ooZXWznl3KAFHo1RA2naZQxXBCqJfOKWMR87MoM0hoVg8NPfEwL9gL5jApW1m3ToMkQRmS8u/PfwEAaM, yjT+gfSMgP3mkMMZ, E0): Missing item digest=U01zZYIUj6YuQIVaH+o/aHIwoKW158dY0ptBESdO9I0=
2022-11-03T06:34:11.770198Z ERROR node{name=0x3d14c61174ec1b9cb34b8a1628d4c7fe389b0b0b}: narwhal_consensus::bullshark: The store does not contain the parent of 9MLghR+os7DOQny/: C1314(q+kxsnIEXHhlRTd3DkOZvx6PIaUyzxPaNZbbQJ2E3QzeZSy9ooZXWznl3KAFHo1RA2naZQxXBCqJfOKWMR87MoM0hoVg8NPfEwL9gL5jApW1m3ToMkQRmS8u/PfwEAaM, v0cg1G6dzlJKcC18, E0): Missing item digest=/DwIDN4SZTV6aaqird5PvqdJ8iyQZahP9kDFoUIx/Kg=
2022-11-03T06:34:23.865163Z  INFO node{name=0x24fe33c99dc8dc906e480b74dfcb1a3df41af0c6}: sui_core::checkpoints: A new checkpoint proposal is created cp_seq=0 size=0
2022-11-03T06:34:23.894597Z  INFO node{name=0x3d14c61174ec1b9cb34b8a1628d4c7fe389b0b0b}: sui_core::checkpoints: A new checkpoint proposal is created cp_seq=0 size=0
```

**Comment by icodezjb** [comment]
the same issue on `sui devnet-v0.15.0`

```txt
2022-11-09T09:13:46.670934Z  INFO sui_cluster_test::faucet: Use local faucet
2022-11-09T09:13:58.357173Z ERROR node{name=0xca503d6ac919288a783d2c64969254a7a71763cd}: narwhal_consensus::bullshark: The store does not contain the parent of ify5WuMrxfNi0s3E: C166(lJc+heYBu+tOAd1vz+FGPjAJF8DezbOhpWuSkeQmLmQGk6MTSNpi2ZwWKmtTbGZaGATeiAw29QePVH7+6UKonycj2d2KamRaLyJj0b5RO59XOIh/iZEyTpbWZwfah6Vu, CG1L3PZUfWp1M5id, E0): Missing item digest=4p98Nzyh3gLvtKqMfbbRvniPkLhpN/HdjP6p76cjyfY=
2022-11-09T09:14:06.430792Z ERROR jsonrpsee_types::params: [next_inner] Deserialization to "core::option::Option<bool>" failed. Error: Error("invalid type: string \"Descending\", expected a boolean", line: 1, column: 12), input JSON: "\"Descending\"]"
2022-11-09T09:14:06.430825Z ERROR sui_json_rpc::api: Error parsing optional "descending_order" as "Option < bool >": InvalidParams(invalid type: string "Descending", expected a boolean at line 1 column 12)
2022-11-09T09:14:06.430872Z ERROR jsonrpsee_types::params: [next_inner] Deserialization to "core::option::Option<bool>" failed. Error: Error("invalid type: string \"Descending\", expected a boolean", line: 1, column: 12), input JSON: "\"Descending\"]"
2022-11-09T09:14:06.430882Z ERROR sui_json_rpc::api: Error parsing optional "descending_order" as "Option < bool >": InvalidParams(invalid type: string "Descending", expected a boolean at line 1 column 12)
2022-11-09T09:14:15.471400Z ERROR node{name=0x617afb76b48399e42500ef9abe6a22816530c803}: narwhal_consensus::bullshark: The store does not contain the parent of aZSgT1WQRxQFTWgU: C370(oCBbTyIk9fjgEJXcD/pwhWYGaZyU9b5BLfO/8Et8V7Gt+O9E+bAYvemUXDJ4yiG4Ccp0ylJDK1uGjHloKZBPhZlQ7bqHG4Mmjl4PJOzUNE36TmK+TutW5iFHdVKeYfvu, uM7Ldt8fT1Y4FRQL, E0): Missing item digest=Jh06s1q1s3a9A/licKzViqLv9yKe9Z80Y5OMcrhF9/U=
2022-11-09T09:14:15.472861Z ERROR node{name=0x4480b90b6ed8c79bdac36e9ab0da512e6606ed2c}: narwhal_consensus::bullshark: The store does not contain the parent of aZSgT1WQRxQFTWgU: C370(oCBbTyIk9fjgEJXcD/pwhWYGaZyU9b5BLfO/8Et8V7Gt+O9E+bAYvemUXDJ4yiG4Ccp0ylJDK1uGjHloKZBPhZlQ7bqHG4Mmjl4PJOzUNE36TmK+TutW5iFHdVKeYfvu, uM7Ldt8fT1Y4FRQL, E0): Missing item digest=Jh06s1q1s3a9A/licKzViqLv9yKe9Z80Y5OMcrhF9/U=
2022-11-09T09:14:15.475192Z ERROR node{name=0xca503d6ac919288a783d2c64969254a7a71763cd}: narwhal_consensus::bullshark: The store does not contain the parent of aZSgT1WQRxQFTWgU: C370(oCBbTyIk9fjgEJXcD/pwhWYGaZyU9b5BLfO/8Et8V7Gt+O9E+bAYvemUXDJ4yiG4Ccp0ylJDK1uGjHloKZBPhZlQ7bqHG4Mmjl4PJOzUNE36TmK+TutW5iFHdVKeYfvu, uM7Ldt8fT1Y4FRQL, E0): Missing item digest=Jh06s1q1s3a9A/licKzViqLv9yKe9Z80Y5OMcrhF9/U=
2022-11-09T09:14:21.613285Z ERROR node{name=0x4480b90b6ed8c79bdac36e9ab0da512e6606ed2c}: narwhal_consensus::bullshark: The store does not contain the parent of XfKk6hoAqae+5w7k: C446(oCBbTyIk9fjgEJXcD/pwhWYGaZyU9b5BLfO/8Et8V7Gt+O9E+bAYvemUXDJ4yiG4Ccp0ylJDK1uGjHloKZBPhZlQ7bqHG4Mmjl4PJOzUNE36TmK+TutW5iFHdVKeYfvu, aDAlNvLkzxgCmFtZ, E0): Missing item digest=yJ+HwE6awyqEL0XHNmEg7g703hBtOc5fj4jD4zOO7SA=
2022-11-09T09:14:47.195353Z ERROR node{name=0x4480b90b6ed8c79bdac36e9ab0da512e6606ed2c}: narwhal_consensus::bullshark: The store does not contain the parent of PBxAZXwn5vHegBjT: C762(lJc+heYBu+tOAd1vz+FGPjAJF8DezbOhpWuSkeQmLmQGk6MTSNpi2ZwWKmtTbGZaGATeiAw29QePVH7+6UKonycj2d2KamRaLyJj0b5RO59XOIh/iZEyTpbWZwfah6Vu, qg6kVMLa9o+Je8mA, E0): Missing item digest=CH8XCBH2a+z8txwfzY8/22G5Jvppd/h/8fjhGwyaLxQ=
2022-11-09T09:14:47.195541Z ERROR node{name=0xca503d6ac919288a783d2c64969254a7a71763cd}: narwhal_consensus::bullshark: The store does not contain the parent of PBxAZXwn5vHegBjT: C762(lJc+heYBu+tOAd1vz+FGPjAJF8DezbOhpWuSkeQmLmQGk6MTSNpi2ZwWKmtTbGZaGATeiAw29QePVH7+6UKonycj2d2KamRaLyJj0b5RO59XOIh/iZEyTpbWZwfah6Vu, qg6kVMLa9o+Je8mA, E0): Missing item digest=CH8XCBH2a+z8txwfzY8/22G5Jvppd/h/8fjhGwyaLxQ=
2022-11-09T09:14:47.195742Z ERROR node{name=0xc63a9f2843fff21f6cce876a12468dcd2e8c8d7a}: narwhal_consensus::bullshark: The store does not contain the parent of PBxAZXwn5vHegBjT: C762(lJc+heYBu+tOAd1vz+FGPjAJF8DezbOhpWuSkeQmLmQGk6MTSNpi2ZwWKmtTbGZaGATeiAw29QePVH7+6UKonycj2d2KamRaLyJj0b5RO59XOIh/iZEyTpbWZwfah6Vu, qg6kVMLa9o+Je8mA, E0): Missing item digest=CH8XCBH2a+z8txwfzY8/22G5Jvppd/h/8fjhGwyaLxQ=
```

**Comment by icodezjb** [comment]
sui-test-validator still doesn't work, is anyone following this issue?

**Comment by icodezjb** [comment]
on sui devnet-v0.17.0 (commit 988e384ef9fdb1f7507ceeae045fa7a6694ef0a0)

- I  can get sui by this cmd, but has some errors.
```bash
curl -H "Content-Type: application/json"     -X POST     -d '{"FixedAmountRequest":{"recipient":"0x82d770bab2d607b919f2dcc45a7491ede65fe6db"}}'     "http://127.0.0.1:9123/gas"
```

erros
```txt
2022-11-25T09:14:37.078450Z  INFO sui_cluster_test::faucet: Use local faucet
2022-11-25T09:14:40.346188Z  INFO sui_faucet::faucet::simple_faucet: Getting faucet requests recipient=0x82d770bab2d607b919f2dcc45a7491ede65fe6db uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.346245Z  INFO sui_faucet::faucet::simple_faucet: Got consumer lock, pulling coins. uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.346259Z  INFO sui_faucet::faucet::simple_faucet: Pulling coin from pool 0x0065c9f6aa5592b3ff06d94702f18a8ab3d4e1c8 uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.347600Z  INFO sui_faucet::faucet::simple_faucet: Planning to use coin from pool 0x0065c9f6aa5592b3ff06d94702f18a8ab3d4e1c8, current balance: 100000000000000 uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.348739Z  INFO sui_faucet::faucet::simple_faucet: PaySui transaction in faucet. tx_digest=ejrHMa4ZpLnxCHgDRqIHLABRXlTF+MIia2ik/CdqpCY= recipient=0x82d770bab2d607b919f2dcc45a7491ede65fe6db coin_id=0x0065c9f6aa5592b3ff06d94702f18a8ab3d4e1c8 uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.476342Z  INFO sui_faucet::faucet::simple_faucet: Got producer lock and recycling gas coin 0x0065c9f6aa5592b3ff06d94702f18a8ab3d4e1c8. uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.476358Z  INFO sui_faucet::faucet::simple_faucet: Recycled coin 0x0065c9f6aa5592b3ff06d94702f18a8ab3d4e1c8 uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06
2022-11-25T09:14:40.476364Z  INFO sui_faucet::faucet::simple_faucet: PaySui txn succeeded in 0.130133346 secs uuid=ce0d68f5-614b-4e6a-835d-ab0d00b98b06 recipient=0x82d770bab2d607b919f2dcc45a7491ede65fe6db digest=ejrHMa4ZpLnxCHgDRqIHLABRXlTF+MIia2ik/CdqpCY=
2022-11-25T09:14:44.107955Z ERROR node{name=0x4805b5e1863a70240b05b4ccd9b8ef52cca7fe72}: narwhal_consensus::bullshark: The store does not contain the parent of 21EPo60LAeVED+TM: C112(uOZntLShIufZpRN4x3SxOwrzKu9E8e8LnxNHDZF5oD6zm6FegSq6N0gNii0kCITLFuH/CbZtcjdyAsjTNyYA7k71c/cpMLYKPYZMQJiwy2txCwu4C41n7ruzcYlgiW/J, 1urW8FZhqF8TGg7/, E0): Missing item digest=UZ+o1Nu519hDTu5b8hwjDVdFzNY1tZ4j0AIsVz21Crw=
2022-11-25T09:14:44.111176Z ERROR node{name=0x3e4835f5f2c61d277ba14d1bf69f4e4fcbe9fc82}: narwhal_consensus::bullshark: The store does not contain the parent of 21EPo60LAeVED+TM: C112(uOZntLShIufZpRN4x3SxOwrzKu9E8e8LnxNHDZF5oD6zm6FegSq6N0gNii0kCITLFuH/CbZtcjdyAsjTNyYA7k71c/cpMLYKPYZMQJiwy2txCwu4C41n7ruzcYlgiW/J, 1urW8FZhqF8TGg7/, E0): Missing item digest=UZ+o1Nu519hDTu5b8hwjDVdFzNY1tZ4j0AIsVz21Crw=
2022-11-25T09:14:44.112436Z ERROR node{name=0x4cddfc3a93464736bb7f47d218aae8f73fbab40e}: narwhal_consensus::bullshark: The store does not contain the parent of 21EPo60LAeVED+TM: C112(uOZntLShIufZpRN4x3SxOwrzKu9E8e8LnxNHDZF5oD6zm6FegSq6N0gNii0kCITLFuH/CbZtcjdyAsjTNyYA7k71c/cpMLYKPYZMQJiwy2txCwu4C41n7ruzcYlgiW/J, 1urW8FZhqF8TGg7/, E0): Missing item digest=UZ+o1Nu519hDTu5b8hwjDVdFzNY1tZ4j0AIsVz21Crw=
2022-11-25T09:14:45.758450Z ERROR node{name=0x4805b5e1863a70240b05b4ccd9b8ef52cca7fe72}: narwhal_consensus::bullshark: The store does not contain the parent of QI6pJ1bY8su0rdZk: C130(jTJvkiJwgfzOvP3KCJPWQTSCD+Omx8HApyB7pQ9kSiJmzOMmv5UGXNpy3RZHcmC9Ayo5tCwcoGqPPOq4DmPYifGGS+djtfminhr0+3FFYs5i1jxWpm+/x2JSK8bzGAPS, ZxQBkLwHo3w7MM07, E0): Missing item digest=SygqwZrhtOwiQ0JZgo0QpOXpBMIPvgcfPVqnv4Q8WFI=
2022-11-25T09:14:45.759622Z ERROR node{name=0x3e4835f5f2c61d277ba14d1bf69f4e4fcbe9fc82}: narwhal_consensus::bullshark: The store does not contain the parent of QI6pJ1bY8su0rdZk: C130(jTJvkiJwgfzOvP3KCJPWQTSCD+Omx8HApyB7pQ9kSiJmzOMmv5UGXNpy3RZHcmC9Ayo5tCwcoGqPPOq4DmPYifGGS+djtfminhr0+3FFYs5i1jxWpm+/x2JSK8bzGAPS, ZxQBkLwHo3w7MM07, E0): Missing item digest=SygqwZrhtOwiQ0JZgo0QpOXpBMIPvgcfPVqnv4Q8WFI=
2022-11-25T09:14:45.763693Z ERROR node{name=0x01b5c47948677ef27c9da5f6535c65759209f051}: narwhal_consensus::bullshark: The store does not contain the parent of QI6pJ1bY8su0rdZk: C130(jTJvkiJwgfzOvP3KCJPWQTSCD+Omx8HApyB7pQ9kSiJmzOMmv5UGXNpy3RZHcmC9Ayo5tCwcoGqPPOq4DmPYifGGS+djtfminhr0+3FFYs5i1jxWpm+/x2JSK8bzGAPS, ZxQBkLwHo3w7MM07, E0): Missing item digest=SygqwZrhtOwiQ0JZgo0QpOXpBMIPvgcfPVqnv4Q8WFI=

```

- I modified the code to change validators from 4 to 3. Looks good without any error!

```bash
$ git diff
diff --git a/crates/sui-cluster-test/src/cluster.rs b/crates/sui-cluster-test/src/cluster.rs
index 86099b5c6..60d5ced13 100644
--- a/crates/sui-cluster-test/src/cluster.rs
+++ b/crates/sui-cluster-test/src/cluster.rs
@@ -140,7 +140,7 @@ impl LocalNewCluster {
 impl Cluster for LocalNewCluster {
     async fn start(options: &ClusterTestOpt) -> Result<Self, anyhow::Error> {
         // Let the faucet account hold 1000 gas objects on genesis
-        let genesis_config = GenesisConfig::custom_genesis(4, 1, 1000);
+        let genesis_config = GenesisConfig::custom_genesis(3, 1, 1000);
 
         // TODO: options should contain port instead of address
         let fullnode_port = options.fullnode_address.as_ref().map(|addr| {
@@ -155,7 +155,9 @@ impl Cluster for LocalNewCluster {
                 .port()
         });
 
-        let mut cluster_builder = TestClusterBuilder::new().set_genesis_config(genesis_config);
+        let mut cluster_builder = TestClusterBuilder::new()
+            .set_genesis_config(genesis_config)
+            .with_num_validators(3);
 
         if let Some(rpc_port) = fullnode_port {
             cluster_builder = cluster_builder.set_fullnode_rpc_port(rpc_port);
diff --git a/crates/sui-config/src/genesis_config.rs b/crates/sui-config/src/genesis_config.rs
index 70eddbedc..3d7eaf27e 100644
--- a/crates/sui-config/src/genesis_config.rs
+++ b/crates/sui-config/src/genesis_config.rs
@@ -280,6 +280,7 @@ impl GenesisConfig {
 
         Self {
             accounts,
+            committee_size: num_authorities,
             ..Default::default()
         }
     }

```

**Comment by icodezjb** [comment]
resolved in sui devnet-v0.21.0
